File: src/main/java/net/snipsniper/configwindow/textpreviewwindow/FolderPreviewRenderer.java
package net.snipsniper.configwindow.textpreviewwindow;

import net.snipsniper.ImageManager;
import org.capturecoop.ccutils.utils.CCStringUtils;

import javax.swing.*;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.util.ArrayList;

public class FolderPreviewRenderer extends JPanel {
    private TextPreviewWindow textPreviewWindow;
    private final BufferedImage folderIcon = ImageManager.Companion.getImage("icons/folder.png");

    public FolderPreviewRenderer(int minWidth, int minHeight) {
        Dimension min = new Dimension(minWidth, minHeight);
        setPreferredSize(min);
        setMinimumSize(min);
    }

    public void setTextPreviewWindow(TextPreviewWindow textPreviewWindow) {
        this.textPreviewWindow = textPreviewWindow;
    }

    @Override
    public void paint(Graphics g) {
        String content = textPreviewWindow.getText().replaceAll("\\\\", "/");
        content = CCStringUtils.formatDateTimeString(content);
        String[] parts = content.split("/");
        ArrayList<String> partsFinal = new ArrayList<>();
        for(String str : parts) {
            if(!str.isEmpty())
                partsFinal.add(str);
        }
        g.setColor(Color.LIGHT_GRAY);
        g.fillRect(0, 0, getWidth(), getHeight());

        partsFinal.add(0, "Main folder");
        int size = getWidth() / (partsFinal.size());
        if(size > getHeight())
            size = getHeight() / (partsFinal.size());
        int index = 0;
        for(String folder : partsFinal) {
            g.drawImage(folderIcon, index * size, 0, size, size, null);
            g.setFont(new Font("Consolas", Font.PLAIN, size/10));
            g.setColor(Color.BLACK);
            g.drawString(folder, (size / 2 - g.getFontMetrics().stringWidth(folder) / 2) + (size * index), (int) (size/1.8));
            index++;
        }
    }

}

File: src/main/kotlin/net/snipsniper/configwindow/textpreviewwindow/FolderPreviewRenderer.kt
package net.snipsniper.configwindow.textpreviewwindow

import net.snipsniper.utils.getImage
import org.capturecoop.ccutils.utils.CCStringUtils
import java.awt.Color
import java.awt.Dimension
import java.awt.Font
import java.awt.Graphics
import javax.swing.JPanel

class FolderPreviewRenderer(width: Int, height: Int): JPanel() {
    private val folderIcon = "icons/folder.png".getImage()
    lateinit var textPreviewWindow: TextPreviewWindow

    init {
        Dimension(width, height).also {
            preferredSize = it
            minimumSize = it
        }
    }

    override fun paint(g: Graphics) {
        val content = CCStringUtils.formatDateTimeString(textPreviewWindow.text.replace("\\\\", "/"))
        val partsFinal = ArrayList<String>()
        content.split("/").filter { it.trim().isNotEmpty() }.forEach { partsFinal.add(it) }

        g.color = Color.LIGHT_GRAY
        g.fillRect(0, 0, width, height)

        partsFinal.add(0, "Main folder")
        var size = width / (partsFinal.size)
        if(size > height) size = height / (partsFinal.size)
        partsFinal.forEachIndexed { index, s ->
            g.drawImage(folderIcon, index * size, 0, size, size, null)
            g.font = Font("Consolas", Font.PLAIN, size/10)
            g.color = Color.BLACK
            g.drawString(s, (size / 2 - g.fontMetrics.stringWidth(s) / 2) + (size * index), (size/1.8).toInt())
        }
    }
}